### PR TITLE
ROX-15135: Reorganize requirements structure and add recommended requirements

### PR DESCRIPTION
**Pull request by mtodor**
Version(s):
RHACS 4.1
**Added 4.2 via cherry pick since this also applies to 4.2 branch**

Issue:
Customers are not able to use existing sizing guides. They are outdated and inaccurate.

Link to docs preview:

- https://61935--docspreview.netlify.app/openshift-acs/latest/installing/acs-high-level-overview.html
- https://61935--docspreview.netlify.app/openshift-acs/latest/installing/acs-default-requirements.html
- https://61935--docspreview.netlify.app/openshift-acs/latest/installing/acs-recommended-requirements.html
- https://61935--docspreview.netlify.app/openshift-acs/latest/cloud_service/acscs-default-requirements.html

QE review:
- [x] QE has approved this change.

**Additional information:**

This PR introduces documentation restructuring and better sizing guides by introducing recommended resources page.

The following restructurings are made:
- Prerequisite pages are moved from setting up sections for different platforms to the upper level because requirements are the same independent of a platform where ACS runs. This is now "Default resource requirements" page.
- High-level overview page is also moved one level up. This is primarily to reflect the moving of Prerequisite pages.

Other changes on existing files:
- links to Prerequisites are now pointing to "Default resource requirements" page
- components are now grouped in single adoc file (central services and secured cluster services)
- High-level overview page has got some modifications to include OpenShift and other platforms. Links are also adjusted.
- Links on Supported platforms page are adjusted
- General requirements page is also changed to include both OpenShift and other platforms instead of using `ifdef`. On the same page `oc` command is removed from the list of required components for Helm installation. Helm does not depend on `oc`.

Added changes:
- Default resource requirements page is added. This is similar to Prerequisites page before. It is organized slightly differently, and values are updated.
- Recommended resource requirements page has been added. This is the primary change and result of perf&scale testing and analysis. This page includes some additional factors that customers should take into account when they are defining requests and limits for our services.

**Considerations**

1. we can revert made restructurings. We just have to find a nice way to incorporate "Recommended resource requirements" page.
Recommended requirements page should be independent of prerequisite/default requirements because it will be improved in the future, and it will be slightly more complex than default requirements.

2. we can split adoc files into files per component as it was before. But this is unnecessary splitting because files will be used together. If we see benefit of reusability in the future, we can split it.

3. if PR is big and it's difficult to review, we can split it into two. One would add restructuring. Another would update default requirements and add recommended requirements.

### CODE REVIEW

File: modules/acs-requirements.adoc
Comment (kcarmichael08) [inline modules/acs-requirements.adoc:4]:
```suggestion
:_content-type: CONCEPT
```

File: modules/acs-requirements.adoc
Comment (kcarmichael08) [inline modules/acs-requirements.adoc:5]:
```suggestion
[id="acs-requirements_{context}"]
```

Comment (kcarmichael08) [reply modules/acs-requirements.adoc:5]: ID needs to mach the module filename (see [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#module-file-metadata)).

**Review by kcarmichael08 — COMMENTED** [review]
I think the ID issue is what is causing it not to build - can you please fix and we can see if the build goes through? That way we will have the HTML preview available also.

**Comment by kcarmichael08** [comment]
Left a comment for why I think the build failed the first time but it looks like it's failing now for a different reason after the latest changes. I can help troubleshoot if you need it, just lmk. :)

File: modules/acs-requirements.adoc
Comment (mtodor) [reply modules/acs-requirements.adoc]: Fixed it! Tnx! I have noticed that we have a mix of `_module-type` and `_content-type` in our repo.

File: modules/default-requirements-secured-cluster-services.adoc
Comment (vikin91) [inline modules/default-requirements-secured-cluster-services.adoc:101]:
```suggestion
| 1 cores
| 500 MiB
```

This is the current state. The memory limit should stay, but the CPU may be lowered (confirmation pending).

File: modules/default-requirements-secured-cluster-services.adoc
Comment (vikin91) [reply modules/default-requirements-secured-cluster-services.adoc:94]: The agreed-on naming variants are: 
- Node Scanning feature
- Node Inventory container

**Review by vikin91 — COMMENTED** [review]
Correcting the first spotted issues. Will add description for those settings in a next round of comments.

File: modules/default-requirements-secured-cluster-services.adoc
Comment (robbycochran) [inline modules/default-requirements-secured-cluster-services.adoc:66]: This block can be removed now that kernel modules are deprecated.

File: modules/default-requirements-secured-cluster-services.adoc
Comment (robbycochran) [inline modules/default-requirements-secured-cluster-services.adoc:113]: This NOTE block can also be removed, it is out of date.

**Review by robbycochran — APPROVED** [review]
Nice work!

File: modules/acs-requirements.adoc
Comment (kcarmichael08) [inline modules/acs-requirements.adoc:34]:
```suggestion
See the default memory and CPU requirements for each component and ensure that the node size can support them.
```

Comment (kcarmichael08) [reply modules/acs-requirements.adoc:34]: Avoid using directional terms like "below," "above," etc. per IBM Style.

File: modules/acs-requirements.adoc
Comment (kcarmichael08) [inline modules/acs-requirements.adoc:1]: Line 40:

- `{ibm-z}` is the updated variable for IBM Z.
- secured cluster services shouldn't be capitalized

Line 51: 

Is the solid state drive bullet supposed to be at the same level as PVC or a sub-bullet of PVC as it is now? (I wasn't sure)

File: modules/default-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/default-requirements-central-services.adoc:15]:
```suggestion
If you are using {rh-rhacscs-first}, you do not need to review the requirements for Central services, because they are managed by Red Hat. You only need to look at the requirements for secured cluster services.
```

File: modules/default-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/default-requirements-central-services.adoc:21]:
```suggestion
A containerized service called Central handles API interactions and {product-title-short} web portal access while a containerized service called Central DB (PostgreSQL 13) handles data persistence.
```

**Review by kcarmichael08 — COMMENTED** [review]
Hi, I'm not done with this, but will break up my review into a couple of parts. Note that before this can be merged (maybe after making all the changes), you will need to squash all commits into one commit.

**Comment by kcarmichael08** [comment]
To clarify comments about "Style" or "Style Guide" - the OpenShift docs team (which we are a part of) has to follow these:

[The IBM Style Guide](https://source.redhat.com/groups/public/ccs/ccs_folder/ibm_style_guide/ibm_stylepdf)
[The Red Hat-specific style guide supplement](https://redhat-documentation.github.io/supplementary-style-guide/)

File: modules/acs-requirements.adoc
Comment (mtodor) [reply modules/acs-requirements.adoc:21]: I think that `one of the following systems` fits better.

File: modules/acs-requirements.adoc
Comment (mtodor) [reply modules/acs-requirements.adoc:34]: WOW! I'm blown away by the word usage list.

File: modules/acs-requirements.adoc
Comment (mtodor) [reply modules/acs-requirements.adoc:20]: I have reorganized this doc a bit to improve bullet points and sections. I'm open to suggestions if you have some ideas on how to improve it.

File: cloud_service/acscs-recommended-requirements.adoc
Comment (kcarmichael08) [inline cloud_service/acscs-recommended-requirements.adoc:12]: This part and the part after the bullets is confusing to me, because it talks about "this test" but there is no information about testing...is this referring to the testing that was done in order to come up with these guidelines? If so, may want to just say something general like "The recommended resource guidelines were developed by performing a test that created the following objects across a given number of namespaces:"

File: cloud_service/acscs-recommended-requirements.adoc
Comment (kcarmichael08) [inline cloud_service/acscs-recommended-requirements.adoc:20]:
```suggestion
During the analysis of results, the number of deployments was identified as a primary factor for increasing of used resources. The number of deployments was used for the estimation of required resources.
```

File: cloud_service/acscs-recommended-requirements.adoc
Comment (kcarmichael08) [reply cloud_service/acscs-recommended-requirements.adoc:12]: Although honestly this is more than the usual level of detail that I've seen given for recommended guidelines, but if you feel that it is important for the user to know this, I defer to your recommendation. Normally we would just say something like "based on our testing, we recommend x."

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:3]: // * cloud_service/acscs-recommended-requirements.adoc

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:16]:
```suggestion
Information for the Collector component is not included on this page. Required resource requirements are listed on the default resource requirements page.
```

Comment (kcarmichael08) [reply modules/recommended-requirements-secured-cluster-services.adoc:16]: We cannot have cross-references in modules, only in assembly files. We can get around this by adding an "Additional resources" section that is tied to the module and that is included in the assembly right after the module is called. The guidelines are here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:27]:
```suggestion
The following table lists the minimum memory and CPU values required to run Sensor on a secured cluster.
```

Comment (kcarmichael08) [reply modules/recommended-requirements-secured-cluster-services.adoc:27]: ^Normally I see Sensor, Scanner, etc. referred to without an article, I think?

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:32]:
```suggestion
| < 25,000
```

Comment (kcarmichael08) [reply modules/recommended-requirements-secured-cluster-services.adoc:32]: https://www.ibm.com/docs/en/ibm-style?topic=measurement-numbers

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:37]:
```suggestion
| < 50,000
```

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:46]:
```suggestion
The Admission controller prevents users from creating workloads that violate policies that you configure.
```

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:56]:
```suggestion
| < 25,000
```

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:61]:
```suggestion
| < 50,000
```

File: installing/acs-recommended-requirements.adoc
Comment (kcarmichael08) [inline installing/acs-recommended-requirements.adoc:9]: Please see comments I made on acscs-recommended-requirements.adoc file

File: modules/default-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/default-requirements-secured-cluster-services.adoc:17]:
```suggestion
Sensor monitors your Kubernetes and {ocp} clusters. These services currently deploy in a single deployment, which handles interactions with the Kubernetes API and coordinates with Collector.
```

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:15]:
```suggestion
If you are using {rh-rhacscs-first}, you do not need to review the requirements for Central services, because they are managed by Red Hat. You only need to look at the requirements for secured cluster services.
```

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:20]:
```suggestion
For default resource requirements for the scanner, see the default resource requirements page.
```

Comment (kcarmichael08) [reply modules/recommended-requirements-central-services.adoc:20]: See previous comment about xrefs in modules & the workaround we need to use if we need a link.

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:29]:
```suggestion
The following table lists the minimum memory and CPU values required to run Central for one secured cluster. The table includes the number of concurrent web portal users.
```

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:34]:
```suggestion
| < 25,000
```

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:39]:
```suggestion
| < 25,000
```

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:44]:
```suggestion
| < 50,000
```

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:49]:
```suggestion
| < 50,000
```

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:61]:
```suggestion
The following table lists the minimum memory and CPU values required for the scanner deployment in the Central cluster. The table includes the number of unique images deployed in all secured clusters.
```

**Review by kcarmichael08 — COMMENTED** [review]
This is really good info and I know it will be super helpful to users. I have some more comments based on Style issues, etc. that are all pretty minor. I would like to take a final look at it after those changes just to make sure all the modules/assemblies have the correct info, but this lgtm otherwise.

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (mtodor) [reply modules/recommended-requirements-secured-cluster-services.adoc:27]: It's mixed. I did a quick search. There are 64 occurrences with `the` and ~180 without.

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (mtodor) [reply modules/recommended-requirements-secured-cluster-services.adoc:16]: I wasn't happy about this reference because it should point to different pages depending on where a module is used. If it's used in cloud_service, then default resources for cloud_service should be linked. If it's in installing, then related default resources in installing tree should be used.

I have changed this a bit. I have removed links from modules and added "Additional resources" in the assembly doc. And in a module, I'm referring to the name used in "Additional resources".

File: cloud_service/acscs-recommended-requirements.adoc
Comment (mtodor) [reply cloud_service/acscs-recommended-requirements.adoc:12]: I have applied your suggestion. I used the word "focused" test.

Inspiration for this page comes from OpenShift: https://docs.openshift.com/container-platform/4.13/scalability_and_performance/recommended-performance-scale-practices/recommended-control-plane-practices.html#master-node-sizing_recommended-control-plane-practices

File: modules/default-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/default-requirements-secured-cluster-services.adoc:1]: These tables look great!

File: modules/recommended-requirements-central-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-central-services.adoc:32]:
```suggestion
| Deployments | Concurrent web portal users | CPU | Memory
```

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:17]:
```suggestion
Collector component is not included on this page. Required resource requirements are listed on the default resource requirements page.
```

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:47]:
```suggestion
The admission controller prevents users from creating workloads that violate policies that you configure.
```

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (kcarmichael08) [inline modules/recommended-requirements-secured-cluster-services.adoc:52]:
```suggestion
The following table lists the minimum memory and CPU values required to run the admission controller on a secured cluster.
```

**Review by kcarmichael08 — COMMENTED** [review]
I think this looks good - I marked a few final things, but it should be OK to squash commits and finalize after these changes! The OpenShift docs repo doesn't allow using "squash and merge" - they [recommend using an interactive rebase](https://docs.google.com/document/u/2/d/e/2PACX-1vRLKWEMHQ3DZroxZKfTu3XcrSdREr6D3oSSayBanEprXhkA2Ciyr2SQuDTYI4aIKUiOPPIQMHgjHeh8/pub#h.epes4xf06pzk) to squash.

File: modules/recommended-requirements-central-services.adoc
Comment (mtodor) [reply modules/recommended-requirements-central-services.adoc:20]: I have moved this note under bullet point. And I have moved the previous note under the title. Because it is actually relevant for the whole Central services section, not only "Central".

I did the same for default requirements.

File: modules/default-requirements-central-services.adoc
Comment (mtodor) [reply modules/default-requirements-central-services.adoc:69]: I have changed it to `Central DB` in the rest of the document. We are using `Central DB` in other documents and it would be better to stay consistent.

File: modules/default-requirements-secured-cluster-services.adoc
Comment (mtodor) [reply modules/default-requirements-secured-cluster-services.adoc:1]: Tnx!

File: modules/recommended-requirements-secured-cluster-services.adoc
Comment (mtodor) [reply modules/recommended-requirements-secured-cluster-services.adoc:17]: I have moved this note under the `Collector` bullet point.

File: modules/default-requirements-central-services.adoc
Comment (mtodor) [reply modules/default-requirements-central-services.adoc:88]: I think this is some copy/paste mistake. I have adjusted it. We should have just a short introduction to Scanner.

**Comment by kcarmichael08** [comment]
Will redirect files with Gaurav's help - [list](https://docs.google.com/document/d/1iBeFkdbOh9BZEqSvgZtaMQhJ7HPj647oQ1kijVH3MCI/edit?usp=sharing)

**Pull request merged**
